src/Microsoft.AspNetCore.Server.IIS/NativeMethods.cs

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Runtime.InteropServices;
+using Microsoft.AspNetCore.HttpSys.Internal;
+
+namespace Microsoft.AspNetCore.Server.IIS
+{
+    public static class NativeMethods
+    {
+        public const int S_OK = 0;
+        private const string AspNetCoreModuleDll = "aspnetcore.dll";
+
+        public enum REQUEST_NOTIFICATION_STATUS
+        {
+            RQ_NOTIFICATION_CONTINUE,
+            RQ_NOTIFICATION_PENDING,
+            RQ_NOTIFICATION_FINISH_REQUEST
+        }
+
+        public delegate REQUEST_NOTIFICATION_STATUS PFN_REQUEST_HANDLER(IntPtr pHttpContext, IntPtr pvRequestContext);
```

Comment:
These need to stay in sync with ANCM. We will also need to expose a version api check here.